### DISCUSSION TITLE
No file format adaptor could read this file.

### DISCUSSION THREAD

**Discussion by daskandalis** [talmolab/sleap#1893] [Help!]
I am trying to run sleap on a cluster, which seems to run correctly until training is initialised.

```
INFO:sleap.nn.training:Loading training labels from: sleap/RL_40-pt_9-cam.pkg.slp
sleap/RL_40-pt_9-cam.pkg.slp doesn't match ext for json or json.zip
Traceback (most recent call last):
  File "/home/usr/.conda/envs/sleap-1.3.3/bin/sleap-train", line 8, in <module>
    sys.exit(main())
  File "/home/usr/.conda/envs/sleap-1.3.3/lib/python3.10/site-packages/sleap/nn/training.py", line 2013, in main
    trainer = create_trainer_using_cli(args=args)
  File "/home/usr/.conda/envs/sleap-1.3.3/lib/python3.10/site-packages/sleap/nn/training.py", line 2000, in create_trainer_using_cli
    trainer = Trainer.from_config(
  File "/home/usr/.conda/envs/sleap-1.3.3/lib/python3.10/site-packages/sleap/nn/training.py", line 666, in from_config
    data_readers = DataReaders.from_config(
  File "/home/usr/.conda/envs/sleap-1.3.3/lib/python3.10/site-packages/sleap/nn/training.py", line 143, in from_config
    return cls.from_labels(
  File "/home/usr/.conda/envs/sleap-1.3.3/lib/python3.10/site-packages/sleap/nn/training.py", line 167, in from_labels
    training: Labels = sleap.load_file(
  File "/home/usr/.conda/envs/sleap-1.3.3/lib/python3.10/site-packages/sleap/io/dataset.py", line 2774, in load_file
    return Labels.load_file(filename, search_paths, match_to=match_to)
  File "/home/usr/.conda/envs/sleap-1.3.3/lib/python3.10/site-packages/sleap/io/dataset.py", line 1968, in load_file
    return read(
  File "/home/usr/.conda/envs/sleap-1.3.3/lib/python3.10/site-packages/sleap/io/format/main.py", line 113, in read
    return disp.read(filename, *args, **kwargs)
  File "/home/usr/.conda/envs/sleap-1.3.3/lib/python3.10/site-packages/sleap/io/format/dispatch.py", line 59, in read
    raise TypeError("No file format adaptor could read this file.")
TypeError: No file format adaptor could read this file.
```

The error might come from the custom install script, if a dependency is missing:

```
conda create -n sleap-1.3.3 python=3.10 -y 
conda activate sleap-1.3.3

wget https://github.com/talmolab/sleap/archive/refs/tags/v1.3.3.tar.gz

tar -zxvf v1.3.3.tar.gz
cd sleap-1.3.3
pip install -r pypi_requirements.txt 
pip install sleap
pip install imgstore==0.2.9 python-dateutil==2.8.2 ruamel.yaml==0.16.13  
pip install pynwb==2.3.3
conda install -n sleap-1.3.3 pyqt -y
pip install segmentation_models

python -c "import sleap; sleap.system_summary()"
```

Any insight would be appreciated. The training package has been built on both Linux and Windows, using settings that work fine locally.

**Comment by eberrigan** [comment]
Hi @daskandalis,

Please include the command you used.

Thanks!

Elizabeth

**Reply by daskandalis** [reply]
Just the regular command:
```
(sleap-1.3.3) [usr@gpu10 USER]$ sleap-train sleap/single_instance.json sleap/RL_40-pt_9-cam.pkg.slp
```
Thanks!

**Reply by daskandalis** [reply]
No obvious cause for this, and it's apparently resolved by repeatedly reinstalling until it works.

**Comment by roomrys** [comment]
Hi @daskandalis,

I am just doing a bit of journaled thinking below, but if you can, would you mind sharing the pkg.slp with us for troubleshooting? If you feel comfortable sharing, we have a [user-upload form](https://docs.google.com/forms/d/e/1FAIpQLSfWBwcuZH8lxGyFabMAODLjSUE4YrSCYbrrnYvXCpgBqZpcJA/viewform).

Thanks,
Liezl

---

>```
>TypeError: No file format adaptor could read this file.
>```

The `LabelsV1Adaptor` should be able to read `.pkg.slp` extensions, but either `LabelsV1Adaptor.can_read_file` is returning `False` sporatically
https://github.com/talmolab/sleap/blob/5ab305c4cf8ef68f815f39071ec0b030ac66d4af/sleap/io/format/hdf5.py#L53-L62

Or `LabelsV1Adaptor` is not being registered in the list of `Dispatch._adaptors`
https://github.com/talmolab/sleap/blob/5ab305c4cf8ef68f815f39071ec0b030ac66d4af/sleap/io/format/dispatch.py#L20-L37
https://github.com/talmolab/sleap/blob/5ab305c4cf8ef68f815f39071ec0b030ac66d4af/sleap/io/format/main.py#L105-L111

**Reply by daskandalis** [reply]
Uploaded an example project that fails to open on my end. I successfully opened it to continue training once, but after that it is failing. Maybe renaming `cfg.outputs.run_name`?

**Comment by roomrys** [comment]
Hi @daskandalis,

I downloaded your data and ran `sleap-train` several times without running into the issue - not great feedback from a debug session. 

I am wondering if this could be a remote training problem?

Things to try would be to run `sleap-train` 
- with absolute paths instead of relative paths
- in the `sleap` directory without any slashes

I am making the assumption that you are running training remotely. If so, I am curious about what version of SLEAP you have on you local computer vs the remote training computer?

Thanks,
Liezl
